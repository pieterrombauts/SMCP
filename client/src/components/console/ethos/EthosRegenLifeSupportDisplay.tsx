import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import UrineProcessor from './UrineProcessor'
import WasteProcessor from './WaterProcessor'
import OxygenGenerator from './OxygenGenerator'
import { generateRandVal } from '../../../dataGenerator';

interface EthosRLSDProps {
    className?: string;
};
const defaultValues = {
    upa: {
        urineTankLevel: 0
    },
    
    wpa: {
        wastewaterTankLevel: 0,
        cleanWaterTankLevel: 0
    },
    
    oga: {
        oxygenProductionRate: 0 
    }
    

}

const UEthosRegenLifeSupportDisplay: React.FC<EthosRLSDProps> = (props) => {
    var [values, setValues] = useState(defaultValues)
    const intervalRef = useRef(0);

    useEffect(() => {
        if (intervalRef.current !== 0) {
            clearInterval(intervalRef.current);
        }
        const id = setInterval(() => {
            var newValues = {
                upa: {
                    urineTankLevel: generateRandVal(60, 62).toFixed(2)
                  },
                
                  wpa: { 
                    wastewaterTankLevel: generateRandVal(60.34, 61.34).toFixed(2),
                    cleanWaterTankLevel: generateRandVal(60.34, 61.34).toFixed(2)
                  },
                
                  oga: {
                    oxygenProductionRate: generateRandVal(60.34, 61.34).toFixed(2)
                  }

            }
            setValues(newValues);
        }, 5000);
        intervalRef.current = id;
    }, [])
    return (
        <div className={props.className}>
            <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
                <image x={0} y={0} width={1300} height={990} href={'/media/Ethos_3.png'}></image>
                <rect x={330} y={390} rx="5" ry="5" width="150" height="150" style={{ fill: "#284b5b" }} />
                <rect x={930} y={310} rx="5" ry="5" width="150" height="150" style={{ fill: "#284b5b" }} />
                <rect x={930} y={750} rx="5" ry="5" width="155" height="150" style={{ fill: "#284b5b" }} />
                <rect x={150} y={810} rx="5" ry="5" width="350" height="100" style={{ fill: "#284b5b" }} />
                

                <UrineProcessor values={values} />
                <WasteProcessor values={values} />
                <OxygenGenerator values={values} />

            </svg>
        </div>
    )
};

const EthosRegenLifeSupportDisplay = styled(UEthosRegenLifeSupportDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
  /* border: 1px solid white; */
`;

export default EthosRegenLifeSupportDisplay;