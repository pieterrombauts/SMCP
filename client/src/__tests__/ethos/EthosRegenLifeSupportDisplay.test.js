import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import EthosRegenLifeSupportDisplay from '../../components/console/ethos/EthosRegenLifeSupportDisplay'
import { shallow, mount, render} from 'enzyme';

describe("Ethos Regen Life Support display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<EthosRegenLifeSupportDisplay />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})