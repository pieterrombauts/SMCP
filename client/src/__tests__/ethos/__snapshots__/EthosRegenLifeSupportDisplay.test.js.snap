// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ethos Regen Life Support display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UEthosRegenLifeSupportDisplay)>
  <UEthosRegenLifeSupportDisplay
    className="c0"
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Ethos_3.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="150"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#284b5b",
            }
          }
          width="150"
          x={330}
          y={390}
        />
        <rect
          height="150"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#284b5b",
            }
          }
          width="150"
          x={930}
          y={310}
        />
        <rect
          height="150"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#284b5b",
            }
          }
          width="155"
          x={930}
          y={750}
        />
        <rect
          height="100"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#284b5b",
            }
          }
          width="350"
          x={150}
          y={810}
        />
        <HarmonyNode
          values={
            Object {
              "oga": Object {
                "oxygenProductionRate": 0,
              },
              "upa": Object {
                "urineTankLevel": 0,
              },
              "wpa": Object {
                "cleanWaterTankLevel": 0,
                "wastewaterTankLevel": 0,
              },
            }
          }
        >
          <g
            id="ethos-thermal-system-display-density-lab"
          >
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="27%"
              y="42%"
            >
              Urine Tank 
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="29%"
              y="45%"
            >
              Level 
            </text>
            <text
              fill="#d6ab62"
              fontFamily="Verdana"
              fontSize="20"
              x="29%"
              y="50%"
            >
              0
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="34%"
              y="50%"
            >
              %
            </text>
          </g>
        </HarmonyNode>
        <WaterProcessor
          values={
            Object {
              "oga": Object {
                "oxygenProductionRate": 0,
              },
              "upa": Object {
                "urineTankLevel": 0,
              },
              "wpa": Object {
                "cleanWaterTankLevel": 0,
                "wastewaterTankLevel": 0,
              },
            }
          }
        >
          <g
            id="ethos-thermal-system-display-density-lab"
          >
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="72.5%"
              y="34.5%"
            >
              Waste Water 
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="73%"
              y="37.5%"
            >
              Tank Level 
            </text>
            <text
              fill="#d6ab62"
              fontFamily="Verdana"
              fontSize="20"
              x="75%"
              y="43%"
            >
              0
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="80%"
              y="43%"
            >
              %
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="72.75%"
              y="78%"
            >
              Clean Water 
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="73%"
              y="81%"
            >
              Tank Level 
            </text>
            <text
              fill="#d6ab62"
              fontFamily="Verdana"
              fontSize="20"
              x="75%"
              y="87%"
            >
              0
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="80%"
              y="87%"
            >
              %
            </text>
          </g>
        </WaterProcessor>
        <OxygenGenerator
          values={
            Object {
              "oga": Object {
                "oxygenProductionRate": 0,
              },
              "upa": Object {
                "urineTankLevel": 0,
              },
              "wpa": Object {
                "cleanWaterTankLevel": 0,
                "wastewaterTankLevel": 0,
              },
            }
          }
        >
          <g
            id="ethos-thermal-system-display-density-lab"
          >
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="17%"
              y="85%"
            >
              O2 Production Rate 
            </text>
            <text
              fill="#d6ab62"
              fontFamily="Verdana"
              fontSize="20"
              x="20%"
              y="89%"
            >
              0
            </text>
            <text
              fill="white"
              fontFamily="Verdana"
              fontSize="20"
              x="25%"
              y="89%"
            >
              kg/day
            </text>
          </g>
        </OxygenGenerator>
      </svg>
    </div>
  </UEthosRegenLifeSupportDisplay>
</Styled(UEthosRegenLifeSupportDisplay)>
`;
